File: py/movieGenre.py
#QUESTION 1
class Movie():
    def __init__(self,g):
        self.uuid = ''
        self.title = ''
        self.year = ''
        self.genres = Genre()
        self.related = []

    #QUESTION 2
    def addRelatedMovie(self,m):
        if(isinstance(m,Movie)):
            self.related.append(m) 
            return True
        return False

    #QUESTION 3
    def setGenre(self,g):
        if(isinstance(g,Genre)):
            if(g.addMovie(self)):
                return True
            return False
        return False 

class Genre():
    def __init__(self):
        self.name = ''
        self.movies = []

    #QUESTION 4  
    def addMovie(self, m):
        if(isinstance(m,Movie)):
            self.movies.append(m)
            return True
        return False

File: js/movieGenre.js
class Movie{
    //QUESTION 1
    constructor (){
        this.uuid = ""
        this.title = ""
        this.year = ""
        this.genre = ""
        this.related = []
    }
    
    //QUESTION 2
    addRelatedMovie(m){
        if(m instanceof Movie){ 
            this.related.push(m)
            return true
        }
        return false
    }

    //QUESTION 3
    setGenre(g){
         if(g instanceof Genre){
            if(g.addMovie(this)){
                return True
            }
            return False
        }
        return False
    } 
}
    
class Genre{
    constructor (){
        this.name = ""
        this.movies = []
    }     

    //QUESTION 4
    addMovie(m){
        if(m instanceof Movie)
            this.movie.push(m)
            return true
        return false
    }
}
    
module.export=
(property).export= {
    Movie: typeof Movie,
    Genre: typeof Genre
}